src/Models/ArduinoDeviceBase.ts

```diff
@@ -232,41 +233,28 @@ export abstract class ArduinoDeviceBase implements Device {
     }
   }
 
-  async generateCrc(
-      context: vscode.ExtensionContext, channel: vscode.OutputChannel) {
+  async generateCrc(channel: vscode.OutputChannel): Promise<void> {
     if (!(vscode.workspace.workspaceFolders &&
           vscode.workspace.workspaceFolders.length > 0)) {
-      const message = 'No workspace opened.';
-      vscode.window.showWarningMessage(message);
-      utils.channelShowAndAppendLine(channel, message);
-      return false;
+      throw new Error('No workspace opened.');
     }
 
     const devicePath = ConfigHandler.get<string>(ConfigKey.devicePath);
     if (!devicePath) {
-      const message = 'No device path found in workspace configuration.';
-      vscode.window.showWarningMessage(message);
-      utils.channelShowAndAppendLine(channel, message);
-      return false;
+      throw new Error('No device path found in workspace configuration.');
     }
     const deviceBuildLocation = path.join(
         vscode.workspace.workspaceFolders[0].uri.fsPath, '..', devicePath,
         '.build');
 
     if (!deviceBuildLocation) {

```

Comment:
Will this variable be null?

Reply:
Should check the directory exists instead of checking whether the path is empty. Fixed.